src/internal/adapter/cast/THEOplayerNativeAirplay.ts

```diff
@@ -1,15 +1,37 @@
-import type { Airplay, CastState, THEOplayerView } from 'react-native-theoplayer';
+import type { Airplay, CastState } from 'react-native-theoplayer';
+import { CastEvent, CastEventType, PlayerEventType, THEOplayer } from 'react-native-theoplayer';
 import { NativeModules } from 'react-native';
 
 export class THEOplayerNativeAirplay implements Airplay {
-  public constructor(private readonly _player: THEOplayerView) {}
+  private readonly _player: THEOplayer;
 
-  casting(): Promise<boolean> {
-    return NativeModules.CastModule.airplayCasting(this._player.nativeHandle);
+  private _casting = false;
+  private _state: CastState = 'available';
+
+  public constructor(player: THEOplayer) {
+    this._player = player;
+    this._player.addEventListener(PlayerEventType.CAST_EVENT, this._onCastStateChange);
+    void this.init_();
+  }
+
+  async init_(): Promise<void> {
+    this._casting = await NativeModules.CastModule.airplayCasting(this._player.nativeHandle);
+    this._state = await NativeModules.CastModule.airplayState(this._player.nativeHandle);
+  }
+
+  private readonly _onCastStateChange = (event: CastEvent) => {
+    if (event.subType === CastEventType.AIRPLAY_STATE_CHANGE) {
+      this._state = event.state;
+      this._casting = event.state === 'connected';
```

Comment:
Just a question: When do we assume that we are 'casting'? Once the state is 'connected' or already when the state is 'connecting'?

Reply:
The cast API for our web SDK also only reporting `casting` as `true` when the state is casting, so I wanted to keep it consistent 👍